app/code/Magento/Persistent/Test/Unit/Helper/SessionTest.php

```diff
@@ -0,0 +1,187 @@
+     */
+    private $sessionFactory;
+
+    /**
+     * @var  \PHPUnit_Framework_MockObject_MockObject|Session
+     */
+    private $session;
+
+    /**
+     * Setup environment
+     */
+    protected function setUp()
+    {
+        $this->context = $this->getMockBuilder(Context::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->dataHelper = $this->getMockBuilder(DataHelper::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->checkoutSession = $this->getMockBuilder(CheckoutSession::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->scopeConfig = $this->getMockBuilder(ScopeConfigInterface::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->sessionFactory = $this->getMockBuilder(SessionFactory::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['create'])
+            ->getMock();
+        $this->session = $this->getMockBuilder(Session::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->sessionFactory->expects($this->any())->method('create')->willReturn($this->session);
+
+        $this->helper = $this->getMockBuilder(SessionHelper::class)
+            ->setMethods(['getSession'])
+            ->setConstructorArgs(
+                [
+                    'context' => $this->context,
+                    'persistentData' => $this->dataHelper,
+                    'checkoutSession' => $this->checkoutSession,
+                    'sessionFactory' => $this->sessionFactory
+                ]
+            )
+            ->getMock();
+    }
+
+    /***
+     * Test isPersistent() when the session has id and enable persistent
+     */
+    public function testIsPersistentWhenSessionId()
+    {
+        $this->session->expects($this->any())->method('getId')
+            ->willReturn(1);
+        $this->helper->expects($this->any())->method('getSession')
+            ->willReturn($this->session);
+        $this->dataHelper->expects($this->any())->method('isEnabled')
+            ->willReturn(true);
+
+        $this->assertEquals(true, $this->helper->isPersistent());
+    }
+
+    /***
+     * Test isPersistent() when the no session id and enable persistent
+     */
+    public function testIsPersistentWhenNoSessionId()
+    {
+        $this->session->expects($this->any())->method('getId')
+            ->willReturn(null);
+        $this->helper->expects($this->any())->method('getSession')
+            ->willReturn($this->session);
+        $this->dataHelper->expects($this->any())->method('isEnabled')
+            ->willReturn(true);
+
+        $this->assertEquals(false, $this->helper->isPersistent());
+    }
+
+    /**
+     * Test isRememberMeChecked() when enable all config
+     */
+    public function testIsRememberMeCheckedWhenEnabledAll()
+    {
+        $testCase = [
```

Comment:
Please use a data provider to pass some test cases, instead of having separate methods for them.

Reply:
It is done. Please check it.